cfg/aws/aws_sdk_logging.go

```diff
@@ -0,0 +1,61 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package aws
+
+import (
+	"log"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/influxdata/wlog"
+)
+
+var sdkLogLevel aws.LogLevelType = aws.LogOff
+
+// SetSDKLogLevel sets the global log level which will be used in all AWS SDK calls.
+// Example usage: set `debug = true` and `aws_sdk_log_level = LogDebug` in config.json.
+func SetSDKLogLevel(wLogLevel wlog.Level, sdkLogLevelString string) {
+	if wLogLevel != wlog.DEBUG {
+		sdkLogLevel = aws.LogOff
+		return
+	}
+	var temp aws.LogLevelType = aws.LogOff
+	// Hard coded strings that match actual variable names in the AWS SDK.
+	// I don't expect these names to change, or their meaning to change.
+	// This code may need updates if the SDK adds more levels (unlikely).
+	// Please note that the levels are a bit field that is OR'd together.
+	if strings.Contains(sdkLogLevelString, "LogDebug") {
```

Comment:
Can you explain this more? I understand that there's some bits being manipulated, but the if statements here concern me.

This `LogDebug` matches all of the other ones below it. Maybe that's not an issue because we're doing an OR, but it smells fishy.

Reply:
https://github.com/aws/amazon-cloudwatch-agent/pull/306#discussion_r743875428